logstash-core/src/main/java/org/logstash/config/source/Base.java

```diff
@@ -0,0 +1,145 @@
+    }
+
+    public boolean isConfigConflict() {
+        throw new UnsupportedOperationException("`isConfigConflict` must be implemented!");
+    }
+
+    //return subclass of LogStash::Setting
+    public IRubyObject configReloadAutomaticSetting() {
+        return this.settings.callMethod(RUBY.getCurrentContext(), "get_setting",
+                RubyString.newString(RUBY, "config.reload.automatic"));
+    }
+
+    public boolean configReloadAutomatic() {
+        final IRubyObject value = configReloadAutomaticSetting().callMethod(RUBY.getCurrentContext(), "value");
+        return value.toJava(Boolean.class);
+    }
+
+    public boolean isConfigReloadAutomatic() {
+        final IRubyObject valueIsSet = configReloadAutomaticSetting().callMethod(RUBY.getCurrentContext(), "set?");
+        return valueIsSet.toJava(Boolean.class);
+    }
+
+    //return subclass of LogStash::Setting
+    public IRubyObject configStringSetting() {
+        return this.settings.callMethod(RUBY.getCurrentContext(), "get_setting",
+                RubyString.newString(RUBY, "config.string"));
+    }
+
+    public String configString() {
+        final IRubyObject setting = configStringSetting();
+        final IRubyObject value = setting.callMethod(RUBY.getCurrentContext(), "value");
+        return value.toJava(String.class);
+    }
+
+    public boolean isConfigString() {
+        return configString() != null;
+    }
+
+    // return subclass of LogStash::Setting
+    public IRubyObject configPathSetting() {
+        return this.settings.callMethod(RUBY.getCurrentContext(), "get_setting",
+                RubyString.newString(RUBY, "path.config"));
+    }
+
+    public String configPath() {
+        return configPathSetting().callMethod(RUBY.getCurrentContext(), "value").toJava(String.class);
+    }
+
+    public boolean isConfigPath() {
+        return !(configPath() == null || configPath().isEmpty());
+    }
+
+    // return subclass of LogStash::Setting
+    public IRubyObject modulesCliSetting() {
+        return this.settings.callMethod(RUBY.getCurrentContext(), "get_setting",
+                RubyString.newString(RUBY, "modules.cli"));
+    }
+
+    @SuppressWarnings("rawtypes")
+    public RubyArray modulesCli() {
+        return modulesCliSetting().callMethod(RUBY.getCurrentContext(), "value").convertToArray();
+    }
+
+    public boolean isModulesCli() {
+        return !(modulesCli() == null || modulesCli().isEmpty());
+    }
+
+    // return subclass of LogStash::Setting
+    public IRubyObject modulesSetting() {
+        return this.settings.callMethod(RUBY.getCurrentContext(), "get_setting",
+                RubyString.newString(RUBY, "modules"));
+    }
+
+    @SuppressWarnings("rawtypes")
+    public RubyArray modules() {
+        return (RubyArray) modulesSetting().callMethod(RUBY.getCurrentContext(), "value");
+    }
+
+    public boolean isModules() {
+        return !(modules() == null || modules().isEmpty());
```

Comment:
with the `modules` parts, as they're being deprecated, think it's fine to have the dynamic "overhead" compared to Ruby